File: examples/native/flutter/lib/screens/settings_screen.dart
import 'package:flutter/material.dart';
import '../tokens/cs_tokens.dart';

class SettingsScreen extends StatefulWidget {
  const SettingsScreen({super.key});

  @override
  State<SettingsScreen> createState() => _SettingsScreenState();
}

class _SettingsScreenState extends State<SettingsScreen> {
  bool dark = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Padding(
          padding: const EdgeInsets.all(24),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.stretch,
            children: [
              Align(
                alignment: Alignment.centerLeft,
                child: TextButton(
                  onPressed: () => Navigator.of(context).pop(),
                  child: Text(
                    '← Back',
                    style: TextStyle(color: CSTokens.colorLink),
                  ),
                ),
              ),
              Text(
                'Settings',
                style: TextStyle(
                  fontSize: 24,
                  fontWeight: FontWeight.w800,
                  color: CSTokens.colorTextPrimary,
                ),
              ),
              const SizedBox(height: 8),
              Text(
                'Third screen — brand swatches from CSTokens.',
                style: TextStyle(fontSize: 13, color: CSTokens.colorTextMuted),
              ),
              const SizedBox(height: 16),
              Container(
                padding: const EdgeInsets.all(16),
                decoration: BoxDecoration(
                  color: CSTokens.colorSurfacePanel,
                  borderRadius: BorderRadius.circular(12),
                  border: Border.all(color: CSTokens.colorBorderDefault),
                ),
                child: Column(
                  children: [
                    SwitchListTile(
                      title: Text(
                        'Prefer dark theme',
                        style: TextStyle(color: CSTokens.colorTextPrimary),
                      ),
                      value: dark,
                      activeColor: CSTokens.colorBrandOchre,
                      onChanged: (v) => setState(() => dark = v),
                    ),
                    _swatch('Brand umber', CSTokens.colorBrandUmber),
                    _swatch('Brand ochre', CSTokens.colorBrandOchre),
                  ],
                ),
              ),
              const Spacer(),
              SizedBox(
                height: CSTokens.componentButtonMdMinHeight,
                child: FilledButton(
                  style: FilledButton.styleFrom(
                    backgroundColor: CSTokens.colorSemanticDanger,
                    foregroundColor: CSTokens.colorTextInverse,
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(
                        CSTokens.componentButtonRadius,
                      ),
                    ),
                  ),
                  onPressed: () =>
                      Navigator.of(context).pushReplacementNamed('/sign-in'),
                  child: const Text('Sign out'),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _swatch(String label, Color color) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 16),
      child: Row(
        children: [
          Expanded(
            child: Text(
              label,
              style: TextStyle(color: CSTokens.colorTextPrimary),
            ),
          ),
          Container(
            width: 36,
            height: 24,
            decoration: BoxDecoration(
              color: color,
              borderRadius: BorderRadius.circular(6),
            ),
          ),
        ],
      ),
    );
  }
}

File: examples/native/compose/app/src/main/java/world/cyberskill/sample/ui/SettingsScreen.kt
package world.cyberskill.sample.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import world.cyberskill.sample.tokens.CSTokens

@Composable
fun SettingsScreen(onSignOut: () -> Unit, onBack: () -> Unit) {
  var dark by remember { mutableStateOf(false) }
  Column(
    modifier = Modifier
      .fillMaxSize()
      .background(CSTokens.colorSurfacePage)
      .padding(24.dp),
    verticalArrangement = Arrangement.spacedBy(16.dp),
  ) {
    TextButton(onClick = onBack) {
      Text("← Back", color = CSTokens.colorLink)
    }
    Text("Settings", color = CSTokens.colorTextPrimary, fontSize = 24.sp)
    Text(
      "Third screen — brand swatches from CSTokens.",
      color = CSTokens.colorTextMuted,
      fontSize = 13.sp,
    )
    Column(
      modifier = Modifier
        .fillMaxWidth()
        .clip(RoundedCornerShape(12.dp))
        .border(1.dp, CSTokens.colorBorderDefault, RoundedCornerShape(12.dp))
        .background(CSTokens.colorSurfacePanel)
        .padding(16.dp),
      verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
      Row(verticalAlignment = Alignment.CenterVertically) {
        Text("Prefer dark theme", color = CSTokens.colorTextPrimary, modifier = Modifier.weight(1f))
        Switch(
          checked = dark,
          onCheckedChange = { dark = it },
          colors = SwitchDefaults.colors(checkedTrackColor = CSTokens.colorBrandOchre),
        )
      }
      SwatchRow("Brand umber", CSTokens.colorBrandUmber)
      SwatchRow("Brand ochre", CSTokens.colorBrandOchre)
    }
    Spacer(modifier = Modifier.weight(1f))
    Button(
      onClick = onSignOut,
      modifier = Modifier
        .fillMaxWidth()
        .height(CSTokens.componentButtonMdMinHeight),
      colors = ButtonDefaults.buttonColors(
        containerColor = CSTokens.colorSemanticDanger,
        contentColor = CSTokens.colorTextInverse,
      ),
      shape = RoundedCornerShape(CSTokens.componentButtonRadius),
    ) {
      Text("Sign out")
    }
  }
}

@Composable
private fun SwatchRow(label: String, color: androidx.compose.ui.graphics.Color) {
  Row(verticalAlignment = Alignment.CenterVertically) {
    Text(label, color = CSTokens.colorTextPrimary, modifier = Modifier.weight(1f))
    Box(
      modifier = Modifier
        .width(36.dp)
        .height(24.dp)
        .clip(RoundedCornerShape(6.dp))
        .background(color),
    )
  }
}
